### PR TITLE
Moved GitLab checks to GitHub's action

### LINKED ISSUE #401: Migrate gitlab checks to github
Let's migrate the CI checks to work on GitHub actions so external contributors can see the results

image

### PR DESCRIPTION
**Pull request by Bullrich**
Resolves #401

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+name: Tests
+
+on:
+  pull_request:
```
Comment (rzadp) [inline .github/workflows/test.yml:4]: Also push to `main` would be nice.

**Review by mutantcornholio — CHANGES_REQUESTED** [review]
Can we have separate checks for lint and test please? 
If check fails, I'd like to know if it's only codestyle or the tests too.

**Comment by Bullrich** [comment]
> Can we have separate checks for lint and test please? If check fails, I'd like to know if it's only codestyle or the tests too.

We have them:
image

`check-lint` checks the lint, and `check-test` checks the test.

If one fails the other one will still run

**Review by mordamax — APPROVED** [review]

**Comment by mutantcornholio** [comment]
I just noticed something: upon submitting a review, the checks were restarted. We should probably select, which `pull_request` events we're listening for

**Comment by Bullrich** [comment]
> I just noticed something: upon submitting a review, the checks were restarted. We should probably select, which `pull_request` events we're listening for

It was a coincidence, I just pushed.

By default, [it only uses `opened`, `synchronized` and `reopened`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) unless you specify.

**Comment by mutantcornholio** [comment]
> > Can we have separate checks for lint and test please? If check fails, I'd like to know if it's only codestyle or the tests too.
> 
> We have them: image
> 
> `check-lint` checks the lint, and `check-test` checks the test.
> 
> If one fails the other one will still run

ah, sorry, my bad. I saw them in the single file, and thought that it'll result in the same check status

**Review by mutantcornholio — APPROVED** [review]

**Review by AndWeHaveAPlan — APPROVED** [review]

**Pull request merged**